content-services/content-sources-frontend
# HMS-5036: Update upload modal to allow cancellation

## Summary

- Adds ability to cancel and restart downloads
- Updates left side file icon to match color
- Left side icon now has a tooltip to explain how this file is being uploaded/if the file data has been reused (pending copy feedback)
- Added an overall upload progress bar.
- Removed the extra spinner to the left side of the toggle dropdown menu (redundant).

![Screenshot 2024-12-19 at 1 40 50 PM](https://github.com/user-attachments/assets/ead0cf27-c657-4e59-bb7e-ca267543258a)
![Screenshot 2024-12-19 at 1 10 07 PM](https://github.com/user-attachments/assets/c82db08b-5e34-4c3b-9f8e-f66edc67a5a5)
![Screenshot 2024-12-19 at 1 00 45 PM](https://github.com/user-attachments/assets/3c4e77de-6a62-4074-a1a2-3fec9ebf13c7)

## Testing steps
- [x] Requires merging of the [PR for HMS-5125 ](https://github.com/content-services/content-sources-backend/pull/918)
- [x] Requires merging of the [PR for HMS-5126](https://github.com/content-services/content-sources-frontend/pull/397)

Once the above are merged (or checked the backend pr 5126 is checked out) 
- Create an upload repository 
- Upload a larger file ([here](https://download-i2.fedoraproject.org/pub/epel/9/Everything/x86_64/Packages/j/?C=S;O=D) is a good example list)
- While it is uploading, hit the cancel button. (can use network tab to turn on slow3g if necessary)
- You can now either close the file, or retry the download (to have it pull the completed chunk list from the backend and restart)

- Any combination of resuming/stopping/closing the modal/opening the modal/ reuploading SHOULD NOT result in any API error response.  If one is seen during testing at any point, let me know.
